scopes/compilation/compiler/compiler.main.runtime.ts

```diff
@@ -27,9 +27,12 @@ export class CompilerMain {
     private builder: BuilderMain
   ) {}
 
+  /**
+   * Run compilation on `bit new` and when new components are imported
```

Comment:
I guess this mainly running when you just run bit compile command

Reply:
not according to @davidfirst and I tend to agree with him from what I could see in the code